feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,753 @@
+			// Fetch CPU utilization data.
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization.Avg == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%", timestamp, deviceName, cpu, cpuParent, cpuUtilization.GetAvg())
+		}
+	}
+
+	if len(lineCards) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", lineCards)
+	}
+}
+
+func TestComponentsNoHighMemoryUtilization(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+	description := "Component"
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cardList := append(controllerCards, lineCards...)
+	for _, component := range cardList {
+		query := gnmi.OC().Component(component).State()
+		timestamp := time.Now().Round(time.Second)
+		componentState := gnmi.Get(t, dut, query)
+		componentType := componentState.GetType()
+		if componentType == lineCardType && deviations.LinecardMemoryUtilizationUnsupported(dut) {
+			t.Logf("INFO: Skipping test for linecard component %s due to deviation linecard_memory_utilization_unsupported", component)
+			continue
+		}
+
+		memoryState := componentState.GetMemory()
+		if memoryState == nil {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Memory data not available",
+				timestamp, deviceName, description, component, componentType)
+			continue
+		}
+
+		memoryAvailable := memoryState.GetAvailable()
+		memoryUtilized := memoryState.GetUtilized()
+		memoryUtilization := uint8((memoryUtilized * 100) / (memoryAvailable + memoryUtilized))
+		t.Logf("INFO: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%%", timestamp, deviceName, description, component, componentType, memoryUtilization)
+	}
+}
+
+func TestSystemProcessNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	const description = "System CPU Process"
+
+	timestamp := time.Now().Round(time.Second)
+	results := gnmi.GetAll(t, dut, query)
+	for _, result := range results {
+		processName := result.GetName()
+
+		if result.CpuUtilization == nil {
+			t.Errorf("%s %s ERROR: %s process %-40s utilization not available", timestamp, deviceName, description, processName)
+		} else {
+			t.Logf("%s %s INFO: %s process %-40s utilization: %3d%%", timestamp, deviceName, description, processName, result.GetCpuUtilization())
+		}
+	}
+}
+
+func TestSystemProcessNoHighMemorySpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	const description = "System Process Memory"
+
+	currentTime := time.Now().Round(time.Second)
+	processes := gnmi.GetAll(t, dut, query)
+	for _, process := range processes {
```

Comment:
t.Run(). Also, is the test supposed to pass if there are no processes?

Reply:
Thanks for your comments.
1. Add t.Run()
```go
	for _, process := range processes {
		processName := process.GetName()
		t.Run(processName, func(t *testing.T) {
```
2. It is supposed to fail.